### DISCUSSION TITLE
Best way to delay startup of PIR HC-SR501

### DISCUSSION THREAD

**Discussion by olonsoft** [arendst/Tasmota#16905] [Q&A] [answered]
I have an HC-SR501 PIR sensor connected to a wemos D1 mini.
According to the datasheet, powering up the HC-SR501 triggers some pulses until the initialization is complete after a minute.
What is the best way for tasmota when powering up, wait a minute and then start sending PIR detections to mqtt?

Is there a better way than the following (that I haven't tested yet if it works) 

```
SwitchMode1 1
SwitchTopic 0

Rule1 on Switch1#state=1 do publish2 stat/%topic%/PIR1 ON endon on Switch1#state=0 do Publish2 stat/%topic%/PIR1 OFF endon

Rule2
  ON System#Init DO Rule1 0 ENDON
  ON System#Boot DO RuleTimer1 60 ENDON
  ON Rules#Timer=1 DO Rule1 1 ENDON

Rule2 1
```

**Comment by sfromis** [comment]
Are you actually experiencing "noise" after power up?

I don't like the method of (in) activating rule sets much. If you really need the delay, you could also assign the value `publish2` or `var9` to var1, and then have `%var1% stat/ ....`. Until you assign `publish2`, the var9 command would effectively be a no-operation. You could also have the whole `publish2 stat/%topic%/PIR1` in a variable the same way, if your %topic% is not too long.

**Comment by olonsoft** [comment, marked as answer]
Thank you for the answer.
Yes, the PIR false triggers while initializing. 

I finally added the following to have a 60 secs delay:

```
SwitchMode1 1
SwitchTopic 0

Rule1
  ON System#Boot DO Var1 tmp ENDON
  ON System#Boot DO RuleTimer1 60 ENDON
  ON Rules#Timer=1 DO Var1 PIR1 ENDON
  ON Switch1#state=1 do publish2 %topic%/stat/%Var1% ON ENDON 
  ON Switch1#state=0 do Publish2 %topic%/stat/%Var1% OFF ENDON

Rule1 1
```

Thank you for the suggestion.
